src/script/descriptor.cpp

```diff
@@ -1491,13 +1491,17 @@ struct KeyParser {
         if (miniscript::IsTapscript(m_script_ctx) && end - begin == 32) {
             XOnlyPubKey pubkey;
             std::copy(begin, end, pubkey.begin());
-            m_keys.push_back(InferPubkey(pubkey.GetEvenCorrespondingCPubKey(), ParseContext(), *m_in));
+            if (auto pubkey_provider = InferPubkey(pubkey.GetEvenCorrespondingCPubKey(), ParseContext(), *m_in)) {
+                m_keys.push_back(std::move(pubkey_provider));
```

Comment:
fb3c5f3e9fb17e2313a6f2aec6db49f3e0da8614 why is `return key` outside the `if` context here, but in the below check it's inside?

Wouldn't that cause it to return `0` instead of `{}`?

Reply:
Fixed